trace/client_test.go

```diff
@@ -349,10 +350,17 @@ func TestReconnectBufferedUNIX(t *testing.T) {
 		mustRecord(t, client, tr)
 
 		assert.NoError(t, <-sentCh)
-		t.Logf("Flushing")
-		go assert.Error(t, Flush(client))
-		// Since reconnections throw away the span, nothing
-		// was received.
+		t.Logf("Flushing to the closed socket")
+		flushErr := make(chan error)
+		// Just keep trying until we get a flush kicked off
+		for FlushAsync(client, flushErr) != nil {
```

Comment:
Is the return value here an error? While it's logically equivalent, `for err := FlushAsync(); err != nil` is slightly more idiomatic.